zulipterminal/model.py

```diff
@@ -896,10 +896,11 @@ def get_current_topic_from_message(
                     return (stream_id, topic)
         return None
 
-    def get_next_unread_topic(self) -> Optional[Tuple[int, str]]:
+    def get_next_unread_topic(self, current_message: int) -> Optional[Tuple[int, str]]:
+        current_topic = self.get_current_topic_from_message(current_message)
         unread_topics = sorted(self.unread_counts["unread_topics"].keys())
         next_topic = False
-        if self._last_unread_topic not in unread_topics:
+        if current_topic is None:
```

Comment:
If we remove the `if current_topic is None` condition and add the `current_topic in unread_topics` condition instead, it will cause existing test cases where `current_topic` is `None` to fail. These test cases represent the case where the first unmuted unread topic is returned. Previously, this behavior was triggered when `last_unread_topic` was not in `unread_topics`. Now, this behavior is triggered when `current_topic` is `None`, indicating that the user is not focused on a stream message. 

> That's why I raised this point, since at a surface level it seems like you're changing/breaking existing behavior in one way and then later fixing it (if in a different way).

To make sure the test cases don't fail, while also ensuring that it doesn't break existing behavior only to fix it in a later commit, one way would be to combine the two commits. I'm not entirely sure if that would be okay, so I'd appreciate any feedback on this. 


> Even better, tests don't cover the existing behavior so would be good to add in a prior commit, even if they demonstrate how unexpected/incorrect the behavior is.

I'll add tests which show the expected behavior in a prior commit, and then change the algorithm in the next commit so as to demonstrate that the changes resolve the incorrect behavior. This will result in the prior commit having some failing test cases, if I understood you correctly?